File: app/MedicalCrab/js/message/message.js


mui.plusReady(function(){
	
	template.helper("imageFormat",function(image){
		return app.Request.ServerUrl+image;
	});
	//与谁聊天
	var self = plus.webview.currentWebview();
	var recivername = self.username;
	
	var user = app.getUser();
	
	var reciver,chat;
	var loading = plus.nativeUI.showWaiting();
	app.api.get("user/userinfo",{username:recivername},function(data){
	    reciver=data.data;
		
		chat=new Chat('ws://192.168.0.109:2014',user.UserName,reciver.UserName); 
		$(".mui-title").html(reciver.UserName); 
		loading.close();
	},function(){
		loading.close();
		self.close();
	});
	

	function Chat(url,username,reciver){
		url=url+"/"+username+"@"+reciver;
		
		this.socket=new WebSocket(url);
		
		var context =this;
		this.socket.onopen=function(){
			console.log("连接到服务器");
			//context.sendTextMessage("wxllzf","hello");
			
		}
		this.socket.onmessage=function(messageEvent){
			var message = JSON.parse(messageEvent.data);
			// alert(message.Content);
			reciveMessage(message.Content);
			// alert(message);
		}
		
		this.socket.onerror=function(){
			
		}
		
		this.sendTextMessage=function(text){
			var username=app.getUser().UserName;
			var message={
				ID:0,
				Sender:username,
				Reciver:reciver,
				Content:text,
				ContentType:0,
				MsgType:0,
				SendTime:"2015/6/9 ",
				ReciveTime:"2015/6/9",
				IsProc:false,
				IsRead:false,
				IsRecived:false
			};
	 
			this.socket.send(JSON.stringify(message));
		}
		
	}
	
	function sendMessage(message){
		var user=app.getUser();
		var data={
			user:user,
			message:{
				content:message
			}
		};
		//ChatLog(message);
		var h = template("template-talk2",data);
	    $(".messages").append(h);
	}
	
	function reciveMessage(message,user){
		var user=app.getUser();
		var data={
			user:reciver,
			message:{
				content:message
			}
		};
		var h = template("template-talk1",data);
	    $(".messages").append(h);
	}
 
    function ChatLog(message){
    	var key=user.UserName+"-"+reciver.UserName;
    	app.storage.setItem(key,message);
    }

	mui(document).on("tap","#btn_send",function(){
		
		var txt =$("#chat_text").val();
		sendMessage(txt);
		chat.sendTextMessage(txt);
	});
});

File: MedicalCrab.Core/Models/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace MedicalCrab.Core.Models
{
    public class Message : Magic.Mvc.Model.Model
    {
        [Display(Name = "消息发送者")]
        public string Sender { get; set; }

        [Display(Name = "消息接收者")]
        public string Reciver { get; set; }

        [Display(Name = "消息内容")]
        public string Content { get; set; }

        [Display(Name = "内容类型")]
        public int ContentType { get; set; }

        [Display(Name = "消息类型")]
        public MsgType MsgType { get; set; }

        [Display(Name = "发送时间")]
        public DateTime SendTime { get; set; }

        [Display(Name = "接收时间")]
        public DateTime ReciveTime { get; set; }

        [Display(Name = "是否处理")]
        public bool IsProc { get; set; }

        [Display(Name = "是否阅读")]
        public bool IsRead { get; set; }

        [Display(Name = "是否接收")]
        public bool IsRecived { get; set; }

    }
}
